### DISCUSSION TITLE
Cant resolve 'fs' when using rainbow kit and next js

### DISCUSSION THREAD

**Discussion by richardryangarcia** [rainbow-me/rainbowkit#1239] [General]
Is anyone seeing the issue "Cant resolve 'fs' " when scaffolding a new rainbowkit and nextjs application? 

I'm trying to try out the new version of rainbowkit with the viem dependency.
I used the recommended command `pnpm create @rainbow-me/rainbowkit@latest` and started up the dev server only to see this error. 

Dependencies in package.json look like this:

`    "@rainbow-me/rainbowkit": "^1.0.0",
    "next": "^12.2.6",
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "viem": "~0.3.19",
    "wagmi": "^1.0.1"`

I could be missing something obvious but thought I'd ask to see if anyone else has experienced this. Thanks for the help!

**Comment by angryhayder** [comment]
nope i spent the whole day looking into it and it's really a problem, seems like a problem with the way next handles server side and client side, i even update to node 20 on mac and still same thing , this is the error message 
ready - started server on 0.0.0.0:3000, url: http://localhost:3000
error - ./node_modules/node-gyp-build/node-gyp-build.js:1:0
Module not found: Can't resolve 'fs'

Import trace for requested module:
./node_modules/node-gyp-build/index.js
./node_modules/bufferutil/index.js
./node_modules/ws/lib/buffer-util.js
./node_modules/ws/lib/sender.js
./node_modules/ws/index.js
./node_modules/@walletconnect/legacy-client/dist/esm/socket.js
./node_modules/@walletconnect/legacy-client/dist/esm/core.js
./node_modules/@walletconnect/legacy-client/dist/esm/index.js
./node_modules/@walletconnect/legacy-provider/dist/esm/signer.js
./node_modules/@walletconnect/legacy-provider/dist/esm/index.js
./node_modules/@wagmi/connectors/dist/walletConnectLegacy.js
./node_modules/@wagmi/core/dist/connectors/walletConnectLegacy.js
./node_modules/wagmi/dist/connectors/walletConnectLegacy.js
./node_modules/@rainbow-me/rainbowkit/dist/index.js
./pages/_app.tsx

https://nextjs.org/docs/messages/module-not-found

**Comment by richardryangarcia** [comment]
Thanks for commenting, at least I know I'm not crazy now. I'm using node 18 and seeing the issue still. Hopefully the team will get back to us about this. Seems like the rainbow kit provider wants to access fs on the client which only exists on the server. something mustve  just changed as this example repo has been up for a while and has always used next.

**Comment by CliveSHD** [comment]
I'm facing the same problem. I've tried various settings, but to no avail. I'm hoping the amazing developers can lend a hand.

**Comment by natedevxyz** [comment]
I had the same problem, and by trial and error I made it work. Don't ask me to explain why though...

In my next.config.js file I added this:

```
webpack: (config, { isServer }) => {
		if (!isServer) {
			config.resolve = {
				...config.resolve,
				fallback: {
					net: false,
					dns: false,
					tls: false,
					path: false,
					fs: false,
				},
			};
		}

		return config;
	}
```

After that, I started getting warnings of missing modules, so I installed them as dev dependencies and it worked.

**Reply by petermazzocco** [reply]
Hey @natedevxyz I tried this out and it threw a bunch of errors related to missing modules, you sure it's just reinstalling them with `--dev` will fix it?

**Reply by natedevxyz** [reply]
It worked for me. Installing them as dev dependencies or regular dependencies made no difference in behavior at first glance. But I think I'll just migrate my project to Vite until libraries like rainbowkit figure out how to work with next 13

Just in case, these were the dependencies I had to install:
- lokijs
- encoding
- pino-pretty

**Comment by petermazzocco** [comment]
Facing the same issue as well, can't figure out what the issue is, but I am also new to Next13 now that it's out of experimental and in production.

I am currently wrapping my Providers in the `layout.tsx` in between the body and wrapping the `{children}`. I'm assuming this is the correct way to go, and I am getting the cannot resolve fs error.

**Reply by angryhayder** [reply]
there's no work around , all fixes for this will throw a bunch of errors somewhere else .
i resorted to building on this repo , it will work fine and when they release a fix you can migrate :
https://github.com/jellydn/wallet-connect-app.
it uses the older pages dir tho
good luck!

**Comment by gtsui** [comment]
No solutions here but just commenting that I'm running into the same issue as well. The issue seems to be stemming from WalletConnect dependencies?

**Comment by petermazzocco** [comment]
Has this been resolved? Or has someone come up with a way to go around it

**Comment by Jam516** [comment]
Had the same issue. 

Fixed by adding this statement to the next.config.js file

module.exports = {
  webpack: (config) => {
    config.resolve.fallback = { fs: false, net: false, tls: false }
    return config
  },
}

Got the answer on connectkit github (connectkit had same issue). Connectkit is nice alternative :) 

https://github.com/family/connectkit/discussions/235

**Comment by DanielSinclair** [comment]
Reference our docs [here](https://www.rainbowkit.com/docs/installation#react-nextjs-and-webpack) and example configuration for Next.js [here](https://github.com/rainbow-me/rainbowkit/tree/main/examples/with-next) to resolve this and install/configure the required polyfills.
